src/actions/sources/loadSourceText.js

```diff
@@ -45,5 +45,6 @@ export function loadSourceText(source: Source) {
     await setSource(newSource);
     await dispatch(setSymbols(source.id));
     await dispatch(setEmptyLines(source.id));
+    await dispatch(setSourceMetaData(source.id));
```

Comment:
@jasonLaster, I think we should keep this like this for now -- it looks fine and we are doing the same thing anyway with setSymbols and setEmptyLines; but we should eventually revisit this, and ask some hard questions. Are these four dispatches truly independent, or are they all concerned with the same piece of data? If we step away from the implementation of loadSourceText; what is loadSourceText doing -- and when is it finished? what is the action to an outside observer (and not to someone who knows how the inside works)?  

I would say this is the best approach for now, and we can change it later on, when we refactor this in earnest and ask these questions. for more information about redux antipatterns: https://github.com/reactjs/redux/issues/857; https://stackoverflow.com/questions/33220776/should-actions-like-showing-hiding-loading-screens-be-handled-by-reducers-to-rel/33226443#33226443

Reply:
Yep... this is definitely an interesting place to investigate. A bit of context:

1. these actions could potentially be slow, so some of the dispatches are staggered to get the data when it's ready
2. some of this is "do the simplest thing" because when we created `actions/ast` it wasn't clear if AST was a separate thing, or if `setPreview` should live w/ pause and `setSymbols` should live with sources...

would love to revisit this too :)